cli/src/parse.rs

```diff
@@ -14,6 +14,22 @@ pub struct Edit {
     pub inserted_text: Vec<u8>,
 }
 
+#[derive(Debug, Default)]
+pub struct Stats {
+    pub successful_parses : usize,
+    pub total_parses : usize,
+}
+
+impl fmt::Display for Stats {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        return writeln!(f, "Total parses: {}; successful parses: {}; failed parses: {}; success percentage: {:.2}%",
```

Comment:
Maybe run `rustfmt` on this file. I think the call would usually be formatted a bit differently than this.

Reply:
`rustfmt` was fine with this line, though it respaced some declarations earlier.